### PR TITLE
Development testtool

### PR DESCRIPTION
**Pull request by BolaNasr**

### CODE REVIEW

**Review by rkhamis — CHANGES_REQUESTED** [review]

File: JumpscaleCore/tools/tester/TesterFactory.py [hunk 1]
```diff
@@ -55,20 +55,24 @@ def test_threebot(self, location):
             test()
 
     def start(self):
+        def run_test(test_name):
+            test_function = j.tools.tester.__getattribute__(test_name)
```
Comment (xmonader) [inline JumpscaleCore/tools/tester/TesterFactory.py]: use `getattr(j.tools.tester, test_name)` instead

File: JumpscaleCore/tools/tester/TesterFactory.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+from unittest import TestCase, skip
+
+from parameterized import parameterized
+
+locations_libs = []
+locations_core = []
+locations_threebot = []
+
+
+class Tester:
```
Comment (xmonader) [inline JumpscaleCore/tools/tester/TesterFactory.py]: No need to have a class here.

File: JumpscaleCore/tools/tester/TesterFactory.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    def _get_all_tests(locations, path):
```
Comment (xmonader) [inline JumpscaleCore/tools/tester/TesterFactory.py]: better to move into a small python function, no need to have that as a staticmethod in a class even (why isn't it decorated with @staticmethod?)

File: JumpscaleCore/tools/tester/TesterFactory.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+        location_search = "__jslocation_"
```
Comment (xmonader) [inline JumpscaleCore/tools/tester/TesterFactory.py]: `__jslocation__`

File: JumpscaleCore/tools/tester/TesterFactory.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        for file_path in j.sal.fs.listFilesInDir(
+            path=j.core.tools.text_replace(path), filter="*Factory.py", recursive=True
+        ):
+            with open(file_path, "r") as f:
```
Comment (xmonader) [inline JumpscaleCore/tools/tester/TesterFactory.py]: j.sal.fs.readFile

File: JumpscaleCore/tools/tester/TesterFactory.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+                content = f.read()
+            if location_search in content:
+                jslocation = content.find(location_search)
+                location = content[content.find("=", jslocation) + 1 : content.find("\n", jslocation)]
```
Comment (xmonader) [inline JumpscaleCore/tools/tester/TesterFactory.py]: better to find with a regular expression

File: JumpscaleCore/tools/tester/TesterFactory.py [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+
+class TesterFactory(j.baseclasses.factory_testtools):
+    __jslocation__ = "j.tools.tester"
+
+    @parameterized.expand(locations_core)
+    def test_core(self, location):
+        module = eval(location)
+        if "test" in dir(module):
+            test = module.__getattribute__("test")
```
Comment (xmonader) [inline JumpscaleCore/tools/tester/TesterFactory.py]: getattr

File: JumpscaleCore/tools/tester/TesterFactory.py [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+            test()
+
+    @parameterized.expand(locations_libs)
+    def test_libs(self, location):
+        module = eval(location)
+        if "test" in dir(module):
+            test = module.__getattribute__("test")
```
Comment (xmonader) [inline JumpscaleCore/tools/tester/TesterFactory.py]: getattr

File: JumpscaleCore/tools/tester/TesterFactory.py [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+            test()
+
+    @parameterized.expand(locations_threebot)
+    def test_threebot(self, location):
+        module = eval(location)
+        if "test" in dir(module):
+            test = module.__getattribute__("test")
```
Comment (xmonader) [inline JumpscaleCore/tools/tester/TesterFactory.py]: getattr

File: JumpscaleCore/tools/tester/TesterFactory.py [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+        def run_test(test_name):
+            test_function = j.tools.tester.__getattribute__(test_name)
+            if test_function:
+                test_function()
+
+        jobs = []
+        j.servers.threebot.local_start_default(background=True)
+
+        for test in dir(j.tools.tester):
+            if not test.endswith("_") and test.startswith("test_"):
```
Comment (xmonader) [inline JumpscaleCore/tools/tester/TesterFactory.py]: why skipping tests ending with `_` ?

**Review by xmonader — CHANGES_REQUESTED** [review]

File: JumpscaleCore/tools/tester/TesterFactory.py (hunk 10 above)
Comment (BolaNasr) [reply JumpscaleCore/tools/tester/TesterFactory.py]: there's one or two tests  the location is empty so the function generated like that ```test_core_29_``` and this is incorrect
anyhow i fixed it

File: JumpscaleCore/tools/tester/TesterFactory.py [hunk 11]
```diff
@@ -0,0 +1,82 @@
+from Jumpscale import j
+
+import os
```
Comment (xmonader) [inline JumpscaleCore/tools/tester/TesterFactory.py]: unused import

**Comment by xmonader** [comment]
closed in favor for the new test runner.

**Pull request closed without merging**